merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/StartOffsetReducer.java

```diff
@@ -104,9 +104,16 @@ private Pair<ActivityDirectiveId, Duration> getNetOffset(ActivityDirective ad){
     boolean anchoredToStart = ad.anchoredToStart();
     Duration netOffset = ad.startOffset();
 
+    final var firstId = currentAnchorId;
+
     while(currentAnchorId != null && anchoredToStart){
       currentActivityDirective = completeMapOfDirectives.get(currentAnchorId);
       currentAnchorId = currentActivityDirective.anchorId();
+
+      if (currentAnchorId.equals(firstId)) {
```

Comment:
This line needs to be reversed to avoid a NPE: `if(firstId.equals(currentAnchorId)`

This is because we know for a fact that `firstId` is not null, as we otherwise wouldn't have entered the while loop, but we expect `currentAnchorId` to become null at some point, assuming there is no cycle.

Reply:
Additionally, I'm not sure if tracking a single id (as opposed to the set of all seen ids so far) is sufficient to detect a cycle, as the cycle may start further up than the initial directive id. If it is, could you please break down _why_ it is?

Reply:
Fixed both